### PR TITLE
feat: AndreiSaroka - task2(Verify Search field functionality) completed

### PR DESCRIPTION
**Pull request by AndreySo1**
Task 2 : Verify Search field functionality
-Created 3 methods to test search functionality.
img1
-Task completed.
-Author: Andrei Saroka

### CODE REVIEW

**Review by Julia864 — COMMENTED** [review]
Test has been reviewed

**Review by Julia864 — APPROVED** [review]

File: demo/src/test/java/amazon/SearchFieldTest.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      PageFactory.initElements(driver, this);
+   }
+
+   @AfterTest
+   public void afterTest() {
+      driver.close();
+      driver.quit();
+   }
+
+   @FindBy(id = "twotabsearchtextbox")
```
Comment (juliaHrabovska) [inline demo/src/test/java/amazon/SearchFieldTest.java:43]: Using @FindBy annotations was expected in the next practice. It is not relevant to have page fields in a test class. Hope to see the PageObject pattern in the next task :)

File: demo/src/test/java/amazon/SearchFieldTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+   @FindBy(xpath = "//div[@data-component-type='s-search-result']//span[@class='a-size-medium a-color-base a-text-normal']")
+   private List<WebElement> searchCardsProduct;
+
+
+   @DataProvider(name = "searchData")
+      public Object[][] searchData() {
+         return new Object[][] {
+                  {"laptop"},
+                  // {"phone"},
```
Comment (juliaHrabovska) [inline demo/src/test/java/amazon/SearchFieldTest.java:60]: Why did you comment on these lines?

File: demo/src/test/java/amazon/SearchFieldTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+   @Test(dataProvider = "searchData")
+   public void verifyResultsSearch(String searchData){
+      searchField.sendKeys(searchData, Keys.ENTER);
+
+      Assert.assertEquals(searchField.getAttribute("value"), searchData); // value search fild contain searchData 
+      Assert.assertTrue(searchRegistr(searchCardsProduct, searchData, false)); // at least card product contains searchData
+   }
+
+   // public boolean searchRegistr(WebElement e, String searchData){
```
Comment (juliaHrabovska) [inline demo/src/test/java/amazon/SearchFieldTest.java:98]: Why this commented code is present here?
it should be removed or uncommented

File: demo/src/test/java/amazon/SearchFieldTest.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+       */
+      boolean result = false;
+      if(list.size()==0) return false;
+
+      String myStr = searchData.substring(0, 1).toUpperCase() + searchData.substring(1);
+
+      for(WebElement element : list){
+         String text = element.getText();
+
+         if(text.matches("^.*"+searchData+".*$") ||
```
Comment (juliaHrabovska) [inline demo/src/test/java/amazon/SearchFieldTest.java:124]: Format your code: use CTRL + ALT + L

File: demo/src/test/java/amazon/SearchFieldTest.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+import org.openqa.selenium.support.PageFactory;
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class SearchFieldTest {
+   protected WebDriver driver;
+   protected String searchData = "laptop"; //default
```
Comment (juliaHrabovska) [inline demo/src/test/java/amazon/SearchFieldTest.java:20]: Not critical, but some comments are redundant

File: demo/src/test/java/amazon/SearchFieldTest.java (hunk 2 above)
Comment (AndreySo1) [reply demo/src/test/java/amazon/SearchFieldTest.java:60]: In task we have one input data. And for faster verification of the verifier.

File: demo/src/test/java/amazon/SearchFieldTest.java (hunk 1 above)
Comment (AndreySo1) [reply demo/src/test/java/amazon/SearchFieldTest.java:43]: ok

File: demo/src/test/java/amazon/SearchFieldTest.java (hunk 3 above)
Comment (AndreySo1) [reply demo/src/test/java/amazon/SearchFieldTest.java:98]: This method have other input parameter, he will need to me or teammates if they will want to use it. But nobody don't want to use now.
And in task not have rule about comments)

File: demo/src/test/java/amazon/SearchFieldTest.java (hunk 5 above)
Comment (AndreySo1) [reply demo/src/test/java/amazon/SearchFieldTest.java:20]: this for my teammates)

File: demo/src/test/java/amazon/SearchFieldTest.java (hunk 4 above)
Comment (AndreySo1) [reply demo/src/test/java/amazon/SearchFieldTest.java:124]: okay

**Pull request merged**